systemtest/src/test/java/io/strimzi/systemtest/SecurityST.java

```diff
@@ -194,13 +196,24 @@ void autoRenewSomeCaCertsTriggeredByAnno(
             boolean zkShouldRoll,
             boolean kafkaShouldRoll,
             boolean eoShouldRoll) throws Exception {
-        createClusterWithExternalRoute();
         String userName = "alice";
-        KafkaUserResource.tlsUser(CLUSTER_NAME, userName).done();
+        int received = 0;
+        String topicName = TOPIC_NAME + "-" + rng.nextInt(Integer.MAX_VALUE);
+
+        createKafkaCluster();
+
+        KafkaUser user = KafkaUserResource.tlsUser(CLUSTER_NAME, userName).done();
+        KafkaTopicResource.topic(CLUSTER_NAME, topicName).done();
+        KafkaClientsResource.deployKafkaClients(true, CLUSTER_NAME + "-" + Constants.KAFKA_CLIENTS, user).done();
+
+        String defaultKafkaClientsPodName =
+                kubeClient().listPodsByPrefixInName(CLUSTER_NAME + "-" + Constants.KAFKA_CLIENTS).get(0).getMetadata().getName();
+
         waitFor("", Constants.GLOBAL_POLL_INTERVAL, Constants.TIMEOUT_FOR_GET_SECRETS, () -> kubeClient().getSecret("alice") != null,
```

Comment:
Yyou can use something like `KafkaUserUtils.waitForSecret()` or `SecretUtils.waitForSecret()` instead of this.